### ISSUE TITLE
Black pixel artifacts when resizing transparent PNG with alternative filters

### ISSUE THREAD

**Issue by vbersier** [gohugoio/hugo#8747]
### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.85.0-724D5DB5+extended linux/amd64 BuildDate=2021-07-05T10:46:28Z VendorInfo=gohugoio
</pre>

### Does this issue reproduce with the latest release?
Yes

### Steps to reproduce

Resize the attached PNG with the parameters `$image.Resize "x800 q85 jpg #ffffff Lanczos"`.

### Comments

When resizing a transparent PNG while adding a background color and using an alternative resizing filter like `Lanczos`, `CatmullRom` or `MitchellNetravali`, there are black pixel artifacts in the output image.

The artifacts are located around the edge of the transparent area, all-around the opaque part of the image.

These artifacts don't show if using `Linear` or `Box` for instance.

Find the original PNG as well as output JPG below.

![lw-lotus-heatshield](https://user-images.githubusercontent.com/64898957/125265343-0c8f6d80-e305-11eb-9b22-d87b56e0e79d.png)

![lw-lotus-heatshield_huf8a62df64b8435dfd1fb0d0cf51b880f_7856480_0x800_resize_q85_bgffffff_lanczos_2](https://user-images.githubusercontent.com/64898957/125265401-1a44f300-e305-11eb-8282-f20c4ac85565.jpg)

**Comment by bep** [comment]
This was fixed in 8f40f34cd10a98598bb822ec633fd5d0ea64b612

**Comment by vbersier** [comment]
Hi. I am not familiar with the inner working of the image processing pipeline, but I did not export to WebP.

Were you able to test with my input image and options? I'll be able once this change is merged into a production version.

Best,
Val

**Comment by bep** [comment]
>but I did not export to WebP. Important to note is that not all filters result in the bug.

This is not only related to WebP (it seems); i just tested your image with your settings before/after that commit.

**Comment by vbersier** [comment]
Sounds good, thanks!

**Comment by vbersier** [comment]
Confirmed fixed! thanks
